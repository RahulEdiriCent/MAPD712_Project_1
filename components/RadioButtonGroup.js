import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';

const RadioButtonGroup = ({ options, selectedOption, onOptionSelect }) => {
  return (
    <View style={styles.container}>
      {options.map((option, index) => (
        <TouchableOpacity
          key={index}
          style={styles.option}
          onPress={() => onOptionSelect(option)}
        >
          <View style={styles.radio}>
            {option === selectedOption && <View style={styles.selectedRadio} />}
          </View>
          <Text style={styles.optionLabel}>{option}</Text>
        </TouchableOpacity>
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  option: {
    flexDirection: 'row',
    alignItems: 'center',
    marginRight: 15
  },
  radio: {
    width: 20,
    height: 20,
    borderRadius: 10,
    borderWidth: 2,
    borderColor: '#6B81DC',
    alignItems: 'center',
    justifyContent: 'center',
  },
  selectedRadio: {
    width: 10,
    height: 10,
    borderRadius: 5,
    backgroundColor: '#6B81DC',
  },
  optionLabel: {
    marginLeft: 5,
    color: '#858585'
  },
});

export default RadioButtonGroup;