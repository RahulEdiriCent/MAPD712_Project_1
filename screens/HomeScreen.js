// HomeScreen.js
import React from 'react';
import { View, Text, Image, StyleSheet, TouchableOpacity, ImageBackground} from 'react-native';
// import LinearGradient from 'react-native-linear-gradient';
// import { Button } from 'react-native-elements';

const HomeScreen = ({ navigation }) => {
  return (
    <ImageBackground source={require('../assets/images/background-image.jpg')} style={styles.backgroundimg}>
      <View style={styles.container}>
        <View style={styles.centerwrapper}>
          <Image source={require('../assets/images/logo.png')} style={styles.logo} />
          {/* <Text style={styles.tagtext}>Wellness at Your Fingertips</Text> */}
          <View style={styles.buttonContainer}>
            <TouchableOpacity style={styles.button} onPress={() => navigation.navigate('AllPatientsScreen')} >
              <Text style={styles.buttonText}>View All Patient</Text>
            </TouchableOpacity> 
            <TouchableOpacity style={[styles.button , styles.nobgbutton]} onPress={() => navigation.navigate('AddPatient')} >
              <Text style={[styles.buttonText, styles.nobgbuttontext]}>Add New Patient</Text>
            </TouchableOpacity> 
          </View>
        </View>
      </View>
    </ImageBackground>
  );
};

const styles = StyleSheet.create({
  backgroundimg: {
    flex: 1,
    resizeMode: 'cover', 
  },
  container: {
    width: '100%',
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  centerwrapper: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  logo: {
    width: 180,
    height: 150,
    marginBottom: 20,
    position: 'absolute',
    top: 150,
  },
  tagtext: {
    fontSize: 18,
    marginBottom: 20,
    marginTop: 30,
  },
  smallImage: {
    height: 90,
    marginTop: 60,
    marginBottom: 80
  },
  buttonContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%',
    position: 'absolute',
    bottom: 50,
  },
  button: {
    backgroundColor: '#6B81DC',
    paddingTop: 12, 
    paddingBottom: 12, 
    borderRadius: 8,
    borderWidth: 1,
    borderColor: '#6B81DC',
    marginBottom: 10,
    marginTop: 10,
    width: 330,
  },
  buttonText: {
    color: '#fff',
    fontSize: 16,
    fontWeight: '600',
    textAlign: 'center'
  },
  nobgbutton: {
    backgroundColor: 'transparent',
  },
  nobgbuttontext: {
    color: '#6B81DC',
    fontWeight: 600,
  }
});
export default HomeScreen;
