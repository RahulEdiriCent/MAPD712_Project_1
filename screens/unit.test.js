import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import RadioButtonGroup from '../components/RadioButtonGroup';
import HomeScreen from './HomeScreen';

// Manually define __DEV__ for testing purposes
global.__DEV__ = true;

describe('HomeScreen', () => {
  // test for radio button
  const options = ['Option 1', 'Option 2', 'Option 3'];
  const selectedOption = 'Option 2';
  let selectedValue = '';

  const onOptionSelect = (option) => {
    selectedValue = option;
  };

  test('renders correctly', () => {
    const { getByText } = render(
      <RadioButtonGroup options={options} selectedOption={selectedOption} onOptionSelect={onOptionSelect} />
    );

    options.forEach((option) => {
      expect(getByText(option)).toBeTruthy();
    });
  });

  test('selects the correct option on press', () => {
    const { getByText } = render(
      <RadioButtonGroup options={options} selectedOption={selectedOption} onOptionSelect={onOptionSelect} />
    );

    options.forEach((option) => {
      const optionButton = getByText(option);
      fireEvent.press(optionButton);
      expect(selectedValue).toBe(option);
    });
  });
  
  // test for homescreen
  test('renders correctly', () => {
    const tree = renderer.create(<HomeScreen navigation={{ navigate: jest.fn() }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('navigates to AllPatientsScreen on "View All Patient" button press', () => {
    const mockNavigation = { navigate: jest.fn() };
    const { getByText } = render(<HomeScreen navigation={mockNavigation} />);
    fireEvent.press(getByText('View All Patient'));
    expect(mockNavigation.navigate).toHaveBeenCalledWith('AllPatientsScreen');
  });

  test('navigates to AddPatient on "Add New Patient" button press', () => {
    const mockNavigation = { navigate: jest.fn() };
    const { getByText } = render(<HomeScreen navigation={mockNavigation} />);
    fireEvent.press(getByText('Add New Patient'));
    expect(mockNavigation.navigate).toHaveBeenCalledWith('AddPatient');
  });
});
