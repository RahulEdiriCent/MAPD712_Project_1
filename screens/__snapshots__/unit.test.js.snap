// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "flex": 1,
      "resizeMode": "cover",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/images/background-image.jpg",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/images/logo.png",
          }
        }
        style={
          {
            "height": 150,
            "marginBottom": 20,
            "position": "absolute",
            "top": 150,
            "width": 180,
          }
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "bottom": 50,
            "justifyContent": "center",
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#6B81DC",
              "borderColor": "#6B81DC",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginBottom": 10,
              "marginTop": 10,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingTop": 12,
              "width": 330,
            }
          }
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "600",
                "textAlign": "center",
              }
            }
          >
            View All Patient
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "transparent",
              "borderColor": "#6B81DC",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginBottom": 10,
              "marginTop": 10,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingTop": 12,
              "width": 330,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#fff",
                  "fontSize": 16,
                  "fontWeight": "600",
                  "textAlign": "center",
                },
                {
                  "color": "#6B81DC",
                  "fontWeight": 600,
                },
              ]
            }
          >
            Add New Patient
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
